app/main.js

```diff
@@ -167,17 +174,84 @@ function createWindow () {
     });
   }
 
+  win.removeAllListeners();
+
+  win.on('close', function(event) {
+    if (minimize) {
+      event.preventDefault();
+      win.hide();
+    }
+  })
+
   win.on('closed', () => {
     win = null
   })
 
+  win.on("hide", () => {
+    if (minimize) updateTray();
+  });
+
+  win.on("show", () => {
+    if (minimize) updateTray();
+  });
+
+  win.on("blur", () => {
+    globalShortcut.unregister('Alt+F4');
+  });
+
+  win.on("focus", () => {
+    globalShortcut.register('Alt+F4', () => safeQuit());
+    if (minimize) updateTray();
+    win.webContents.send('window-is-focussed', null);
+  });
+
   // Menu bar
   win.setAutoHideMenuBar(true);
   win.setMenuBarVisibility(isDebug);
   setMenu();
 
 };
 
+function createTray () {
+  // Minimize to tray logic follows:
+  // Set the tray icon
+  const iconPath = path.join(app.getAppPath(), "/dist/img/fav/32x32.png");
+  tray = new Tray(iconPath);
+  tray.setToolTip("LBRY App");
+  tray.setTitle("LBRY");
+}
+
+// This needs to be done as for linux the context menu doesn't update automatically(docs)
+function updateTray() {
+  let contextMenu = Menu.buildFromTemplate(getMenuTemplate());
+  tray.setContextMenu(contextMenu);
+}
+
+function getMenuTemplate () {
+  return [
+    getToggleItem(),
+    {
+      label: "Quit",
+      click: () => safeQuit(),
+    },
+  ]
+
+  function getToggleItem () {
+    if (win.isVisible()) {
```

Comment:
I think this needs to be a check if the window is focused, not just if it is visible. If I Alt+Tab from an open app, I am shown "Hide" in the context menu, when it should be "Show".

Reply:
Done